plugins/modules/inventory_intent.py

```diff
@@ -795,6 +804,38 @@ def validate_input(self):
 
         return self
 
+    def get_device_ips_from_config_priority(self):
+        """
+        Retrieve device IPs based on the configuration.
+        Args:
+            -  self (object): An instance of a class used for interacting with Cisco Cisco Catalyst Center.
+        Returns:
+            list: A list containing device IPs.
+        Description:
+            This method retrieves device IPs based on the priority order specified in the configuration.
+            It first checks if device IPs are available. If not, it checks hostnames, serial numbers,
+            and MAC addresses in order and retrieves IPs based on availability.
+            If none of the information is available, an empty list is returned.
+        """
+        device_ips = self.config[0].get("ip_address")
+
+        if device_ips:
+            return device_ips
+
+        device_hostnames = self.config[0].get("hostname_list")
+        if device_hostnames:
+            return self.get_device_ips_from_hostname(device_hostnames)
+
```

Comment:
`    # If hostnames are not available, check serial numbers
`

Reply:
added the comment.